src/main/java/com/thoughtworks/gauge/scan/CustomClassInitializerScanner.java

```diff
@@ -17,29 +15,20 @@ public void scan(Reflections reflections) {
     }
 
     private void scanForInitializer(Reflections reflections) {
-        Set<Class<?>> initializers = reflections.getTypesAnnotatedWith(CustomClassInitializer.class);
-
-        if (initializers.isEmpty()) {
-            ClassInitializerRegistry.classInitializer(new DefaultClassInitializer());
-            return;
-        }
-
-        if (initializers.size() > 1) {
-            throw new ScanningException("Only 1 class can be annotated with @CustomClassInitializer");
-        }
-
-        try {
-            Class<?> initializer = initializers.iterator().next();
-            if (!ClassInitializer.class.isAssignableFrom(initializer)) {
-                throw new ScanningException("Custom class initializer should implement ClassInitializer interface");
+        Set<Class<? extends ClassInitializer>> initializers = reflections.getSubTypesOf(ClassInitializer.class);
+        initializers.remove(DefaultClassInitializer.class);
+        if (initializers.size() == 1) {
+            Class<? extends ClassInitializer> initializer = initializers.iterator().next();
+            try {
+                ClassInitializerRegistry.classInitializer(initializer.newInstance());
+                System.out.println(String.format("Using %s as class initializer", initializer.getName()));
+            } catch (InstantiationException e) {
+                System.err.println(String.format("Could not instantiate %s, continuing using default class initializer", initializer.getName()));
+            } catch (IllegalAccessException e) {
+                System.err.println(String.format("Could not access %s constructor, continuing using default class initializer", initializer.getName()));
             }
-            ClassInitializerRegistry.classInitializer((ClassInitializer) initializer.newInstance());
-        } catch (InstantiationException e) {
-            throw new ScanningException(e.getMessage());
-        } catch (IllegalAccessException e) {
-            throw new ScanningException(e.getMessage());
-        } catch (NoSuchElementException e) {
-            // Class initializer is not defined, use default!
+        } else {
```

Comment:
There can be three cases here
* initializers == 0 [custom class initialiser not found]
* initializers == 1 [single custom class initialiser found]
* initializers > 1 [multiple custom class initialiser found]

We should just show the message for the last condition. But this code will print for first and third conditions both.